src/client/pfs/pfs.proto

```diff
@@ -495,6 +500,17 @@ message GetTarRequest {
   File file = 1;
 }
 
+message GetTarConditionalRequest {
+  File file = 1;
+  bool skip = 2;
+}
+
+message GetTarConditionalResponse {
+  FileInfoNewStorage file_info = 1;
+  bytes data = 2;
+  bool done = 3;
+}
+
```

Comment:
Could you add some comments here for which fields are going to be set simultaneously, I think most of these fields are mutually exclusive, but I'm not sure.

Reply:
Yeah, I am debating where I should document the protocol in the code. I was thinking somewhere in the proto file, but I was not sure if it should be near the messages or rpc definition. Also, I was waiting to see if there was any feedback that would result with changes to the protocol.

Reply:
Totally reasonable to wait until things feel settled before documenting them. My general philosophy is that for comments like this that define data structure invariants the best place to put them is where the data structure is defined, even though that's not normally where the constraint is being enforced, I think of the definition as the "source of truth" for the data structure which in my mind includes invariants like this. Also practically speaking putting it there increases the chances that if the data structure gets updated in some way the comment will be updated to reflect it. Also I'm totally on board with the "why not both" approach, if there's other places where spelling out this invariant feels like it clarifies things feel free to add it there.